Adding W solid properties materials

meltawila opened the pull request:
ref #31072 

@simopier @GiudGiud

Commit:
Adding tungsten thermal conductivity

Commit:
Adding tungsten thermal properties

Removing trailing whitespace

GiudGiud commented on modules/solid_properties/test/tests/materials/tungsten/tests:
```diff
@@ -0,0 +1,10 @@
+  [diffs]
+    type = Exodiff
+    input = 'tungsten_thermal_properties.i'
+    exodiff = 'tungsten_thermal_properties_out.e'
```
use a LineValueSampler and a CSVDiff. Will be easier to check the results

GiudGiud reviewed (changes requested):
thanks for the PR, this is on a good path

Commit:
Address Guillaume's comments (idaholab#31072)

meltawila commented on modules/solid_properties/src/materials/TungstenThermalPropertiesMaterial.C:
```diff
@@ -0,0 +1,67 @@
+    const InputParameters & parameters)
+  : Material(parameters),
+    _temperature(coupledValue("temperature")),
+    _k(declareProperty<Real>(
```
I found an example here with GenericMaterialProperty<Real, is_ad> and followed it. I checked that my code works for both AD and non AD variations https://mooseframework.inl.gov/automatic_differentiation/templated_objects.html

joshuahansel reviewed (dismissed):
What is the reason to make a new Material instead of a new SolidProperties and use it with `ThermalSolidPropertiesMaterial`? @GiudGiud

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,17 @@
+
+!syntax description /Materials/TungstenThermalPropertiesMaterial
+
+`TungstenThermalPropertiesMaterial` provides tungsten's thermal conductivity, specific heat, and density as material properties based on temperature
```
```suggestion
This class provides tungsten's thermal conductivity, specific heat, and density as material properties based on temperature.
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,17 @@
+
+`TungstenThermalPropertiesMaterial` provides tungsten's thermal conductivity, specific heat, and density as material properties based on temperature
+
+Properties of tungsten are obtained from [!cite](milner2024space).
```
Please provide the expression for each property, along with the units and the temperature range of validity.

simopier commented on modules/solid_properties/include/materials/TungstenThermalPropertiesMaterial.h:
```diff
@@ -0,0 +1,75 @@
+  virtual void computeQpProperties();
+
+  // Constants for the thermal conductivity formulas
+  static constexpr Real _kA0 = 7.348e+05;
+  static constexpr Real _kA1 = 2.544e+01;
+  static constexpr Real _kA2 = -8.304e+03;
+  static constexpr Real _kA3 = 1.180e+06;
+  static constexpr Real _kB0 = -3.679;
+  static constexpr Real _kB1 = 1.181e+02;
+  static constexpr Real _kB2 = 5.879e+01;
+  static constexpr Real _kB3 = 2.867;
+  static constexpr Real _kC0 = -2.052e-02;
+  static constexpr Real _kC1 = 4.741e-01;
+
+  // Constants for the specific heat formulas
+  static constexpr Real _cN = 3.030;
+  static constexpr Real _cA0 = 3.103e+02;
+  static constexpr Real _cA1 = -8.815;
+  static constexpr Real _cA2 = 1.295e+02;
+  static constexpr Real _cA3 = 1.874e+03;
+  static constexpr Real _cB0 = 1.301e-01;
+  static constexpr Real _cB1 = 2.225e-02;
+  static constexpr Real _cB2 = -7.224e-03;
+  static constexpr Real _cB3 = 3.539e-03;
+  static constexpr Real _cB_2 = -3.061e-04;
+  // Constatns for the density formulas
+  static constexpr Real _rA0 = 19250;
+  static constexpr Real _rA1 = -8.529e-02;
+  static constexpr Real _rA2 = -9.915e-02;
+  static constexpr Real _rA3 = 2.257;
+  static constexpr Real _rA4 = -3.157;
+  static constexpr Real _rB0 = -1.4e-01;
+  static constexpr Real _rB1 = 4.869e-01;
+  static constexpr Real _rB2 = -3.056e-02;
+  static constexpr Real _rB3 = 2.234e-02;
```
Please add comments with units on each line.

simopier commented on modules/solid_properties/src/materials/TungstenThermalPropertiesMaterial.C:
```diff
@@ -0,0 +1,94 @@
+{
+  if (_temperature[_qp] < 11 || _temperature[_qp] > 3600)
+    flagInvalidSolution("Thermal properties out of bounds for the Tungsten temperature "
+                        "computation. Temperature has to be between 11K and 3600K.");
```
```suggestion
                        "computation. Temperature has to be between 11 K and 3600 K.");
```

simopier commented on modules/solid_properties/src/materials/TungstenThermalPropertiesMaterial.C:
```diff
@@ -0,0 +1,94 @@
+TungstenThermalPropertiesMaterialTempl<is_ad>::computeQpProperties()
+{
+  if (_temperature[_qp] < 11 || _temperature[_qp] > 3600)
+    flagInvalidSolution("Thermal properties out of bounds for the Tungsten temperature "
```
@GiudGiud what does `flagInvalidSolution` do? 
Is it an error? Does it cut the time step? Does it provide a warning? 
I'm asking because being able to control that behavior is key in some cases. Here, the range seems to be quite large, but in some cases, it is important to either be intolerant or flexible whenever the variable gets out of bounds.

simopier commented on modules/solid_properties/src/materials/TungstenThermalPropertiesMaterial.C:
```diff
@@ -0,0 +1,94 @@
+  if (_temperature[_qp] < 11 || _temperature[_qp] > 3600)
+    flagInvalidSolution("Thermal properties out of bounds for the Tungsten temperature "
+                        "computation. Temperature has to be between 11K and 3600K.");
+
```
It might be worth defining `temperature_scaled = _temperature[_qp] / 1000` here given how many times this is used.

simopier commented on modules/solid_properties/src/materials/TungstenThermalPropertiesMaterial.C:
```diff
@@ -0,0 +1,94 @@
+void
+TungstenThermalPropertiesMaterialTempl<is_ad>::computeQpProperties()
+{
+  if (_temperature[_qp] < 11 || _temperature[_qp] > 3600)
```
Based on Milner et al., this range is not valid for all these properties. Or rather, this range is valid for all these properties, but you are restricting the ranges of all three properties to the smallest common range. 
Ranges can be wider for some of these properties, and this should be reflected in this class. 
They should all have their own range that differs slightly.

simopier commented on modules/solid_properties/test/tests/materials/tungsten/tests:
```diff
@@ -0,0 +1,10 @@
+[Tests]
```
I think you should add a test for when the class is used outside the range of definition of the properties.

simopier reviewed (changes requested):
Great contribution @meltawila! 

I have some comments for you to address. Mostly, I want to make sure things are well documented, and I also want to make sure the range of applicability is explicit to each property, and can be used flexibly.

joshuahansel commented:
> at this point, can you use a Material in THM or does it need to be a SolidProperties UO?

Yes, you can use Materials directly.

Commit:
Adding equations to docs and make flagInvalidSolution explicit for each property

Commit:
adding invalid temperature test and fixing doc

Apply suggestions from code review

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

fix bad doc formatting

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/3066335/), step Docs: sync website on 4bc6486 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/31073/site)

This comment will be updated on new commits.

meltawila committed:
Address Guillaume's reivew:
- add units on temperature in doco
- start AD templating of matprop
- remove tests for invalid-solution behavior

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

meltawila committed:
include headers, remove invalidity tests, and update .md

meltawila committed:
use coupledGenericValue

simopier commented:
Please let me review this one more time before merging.

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+!syntax description /Materials/TungstenThermalPropertiesMaterial
+
+This class provides tungsten's thermal conductivity, specific heat, and density as material properties based on temperature.
+
+Properties of tungsten are obtained from [!cite](milner2024space).
```
```suggestion
This class provides tungsten's thermal conductivity, specific heat, and density as material properties as a function of temperature based on the models listed in [!cite](milner2024space).
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+\begin{equation}
+k(T) =
+\frac{A0 \cdot \left( \frac{T}{1000} \right)^N}{1 + A1 \cdot \left( \frac{T}{1000} \right) + A2 \cdot \left( \frac{T}{1000} \right)^2 + A3 \cdot \left( \frac{T}{1000} \right)^3}
```
```suggestion
\frac{A0 \cdot \left( \frac{T}{1000} \right)^N}{1 + A1 \cdot \left( \frac{T}{1000} \right) + A2 \cdot \left( \frac{T}{1000} \right)^2 + A3 \cdot \left( \frac{T}{1000} \right)^3},
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+where:
+
+- `k(T)` is **thermal conductivity `[W/m.K]`
+
+- `T` is **temperature `[K]`
```
```suggestion
- $k(T)$ is the thermal conductivity [W/m.K],

- $T$ is the temperature [K]`,
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+For temperature range: 1 ≤ T < 55 K
+
+\begin{equation}
```
```suggestion
\begin{equation} \label{eq:thermal_conductivity_low}
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+# Thermal conductivity
+
+For temperature range: 1 ≤ T < 55 K
```
```suggestion
For temperature range: 1 ≤ T < 55 K, the thermal conductivity is defined as
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+- `k(T)` is **thermal conductivity `[W/m.K]`
+
+- `T` is **temperature `[K]`
+
```
```suggestion

and the model parameters values provided in [tab:thermal_conductivity_parameters_low].

!table id=tab:thermal_conductivity_parameters_low caption=Parameters values for the thermal conductivity model from [eq:thermal_conductivity_low]. 
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+| Constant | Value      | Unit                |
+|----------|------------|---------------------|
+| N        | 8.740E-01  | `[1]`               |
+| A0       | 7.348E+05  | `[W/m.K]`           |
+| A1       | 2.544E+01  | `[1]`               |
+| A2       | -8.304E+03 | `[1]`               |
+| A3       | 1.180E+06  | `[1]`               |
```
```suggestion
| N        | 8.740E-01  | `[-]`               |
| A0       | 7.348E+05  | `[W/m.K]`           |
| A1       | 2.544E+01  | `[-]`               |
| A2       | -8.304E+03 | `[-]`               |
| A3       | 1.180E+06  | `[-]`               |
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+For temperature range 55 ≤ T ≤ 3653 K
+
+\begin{equation}
```
```suggestion
\begin{equation} [eq:thermal_conductivity_high]
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+For temperature range 55 ≤ T ≤ 3653 K
+
+\begin{equation}
+k(T) = \frac{B0 + B1 \cdot \left( \frac{T}{1000} \right) + B2 \cdot \left( \frac{T}{1000} \right)^2 + B3 \cdot \left( \frac{T}{1000} \right)^3}{C0 + C1 \cdot \left( \frac{T}{1000} \right) + \left( \frac{T}{1000} \right)^2}
```
```suggestion
k(T) = \frac{B0 + B1 \cdot \left( \frac{T}{1000} \right) + B2 \cdot \left( \frac{T}{1000} \right)^2 + B3 \cdot \left( \frac{T}{1000} \right)^3}{C0 + C1 \cdot \left( \frac{T}{1000} \right) + \left( \frac{T}{1000} \right)^2},
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+| A2       | -8.304E+03 | `[1]`               |
+| A3       | 1.180E+06  | `[1]`               |
+
+For temperature range 55 ≤ T ≤ 3653 K
```
```suggestion
For temperature range 55 ≤ $T$ ≤ 3653 K, the thermal conductivity is defined as
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+where:
+
+- `k(T)` is **thermal conductivity `[W/m.K]`
+
+- `T` is **temperature `[K]`
+
+| Constant | Value       | Unit                |
```
```suggestion
- $k(T)$ is the thermal conductivity [W/m.K],
- $T$ is the temperature [K]`,

and the model parameters values provided in [tab:thermal_conductivity_parameters_high].

!table id=tab:thermal_conductivity_parameters_high caption=Parameters values for the thermal conductivity model from [eq:thermal_conductivity_high]. 

| Constant | Value       | Unit                |
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+| B1       | 1.181E+02   | `[W/m.K]`           |
+| B2       | 5.879E+01   | `[W/m.K]`           |
+| B3       | 2.867E+00   | `[W/m.K]`           |
+| C0       | -2.052E-02  | `[1]`               |
+| C1       | 4.741E-01   | `[1]`               |
```
```suggestion
| C0       | -2.052E-02  | `[-]`               |
| C1       | 4.741E-01   | `[-]`               |
```

simopier commented on modules/solid_properties/include/materials/TungstenThermalPropertiesMaterial.h:
```diff
@@ -0,0 +1,75 @@
+
+  // Constants for the thermal conductivity formulas
+  static constexpr Real _kA0 = 7.348e+05;  // [W/m.K]
+  static constexpr Real _kA1 = 2.544e+01;  // [1]
+  static constexpr Real _kA2 = -8.304e+03; // [1]
+  static constexpr Real _kA3 = 1.180e+06;  // [1]
```
```suggestion
  static constexpr Real _kA1 = 2.544e+01;  // [-]
  static constexpr Real _kA2 = -8.304e+03; // [-]
  static constexpr Real _kA3 = 1.180e+06;  // [-]
```

simopier commented on modules/solid_properties/include/materials/TungstenThermalPropertiesMaterial.h:
```diff
@@ -0,0 +1,75 @@
+  static constexpr Real _kB1 = 1.181e+02;  // [W/m.K]
+  static constexpr Real _kB2 = 5.879e+01;  // [W/m.K]
+  static constexpr Real _kB3 = 2.867;      // [W/m.K]
+  static constexpr Real _kC0 = -2.052e-02; // [1]
+  static constexpr Real _kC1 = 4.741e-01;  // [1]
```
```suggestion
  static constexpr Real _kC0 = -2.052e-02; // [-]
  static constexpr Real _kC1 = 4.741e-01;  // [-]
```

simopier commented on modules/solid_properties/include/materials/TungstenThermalPropertiesMaterial.h:
```diff
@@ -0,0 +1,75 @@
+  static constexpr Real _kC1 = 4.741e-01;  // [1]
+
+  // Constants for the specific heat formulas
+  static constexpr Real _cN = 3.030;        // [1]
+  static constexpr Real _cA0 = 3.103e+02;   // [J/g.K]
+  static constexpr Real _cA1 = -8.815;      // [1]
+  static constexpr Real _cA2 = 1.295e+02;   // [1]
+  static constexpr Real _cA3 = 1.874e+03;   // [1]
```
```suggestion
  static constexpr Real _cN = 3.030;        // [-]
  static constexpr Real _cA0 = 3.103e+02;   // [J/g.K]
  static constexpr Real _cA1 = -8.815;      // [-]
  static constexpr Real _cA2 = 1.295e+02;   // [-]
  static constexpr Real _cA3 = 1.874e+03;   // [-]
```

simopier commented on modules/solid_properties/include/materials/TungstenThermalPropertiesMaterial.h:
```diff
@@ -0,0 +1,75 @@
+  static constexpr Real _cB_2 = -3.061e-04; // [J/g.K]
+  // Constatns for the density formulas
+  static constexpr Real _rA0 = 19250;      // [Kg/m^3]
+  static constexpr Real _rA1 = -8.529e-02; // [1]
+  static constexpr Real _rA2 = -9.915e-02; // [1]
+  static constexpr Real _rA3 = 2.257;      // [1]
+  static constexpr Real _rA4 = -3.157;     // [1]
+  static constexpr Real _rB0 = -1.4e-01;   // [1]
+  static constexpr Real _rB1 = 4.869e-01;  // [1]
+  static constexpr Real _rB2 = -3.056e-02; // [1]
+  static constexpr Real _rB3 = 2.234e-02;  // [1]
```
```suggestion
  static constexpr Real _rA1 = -8.529e-02; // [-]
  static constexpr Real _rA2 = -9.915e-02; // [-]
  static constexpr Real _rA3 = 2.257;      // [-]
  static constexpr Real _rA4 = -3.157;     // [-]
  static constexpr Real _rB0 = -1.4e-01;   // [-]
  static constexpr Real _rB1 = 4.869e-01;  // [-]
  static constexpr Real _rB2 = -3.056e-02; // [-]
  static constexpr Real _rB3 = 2.234e-02;  // [-]
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+For temperature range: 11 ≤ T < 293 K
+
+\begin{equation}
+C_p(T) =\frac{A0 \cdot \left( \frac{T}{1000} \right)^N}{1 + A1 \cdot \left( \frac{T}{1000} \right) + A2 \cdot \left( \frac{T}{1000} \right)^2 + A3 \cdot \left( \frac{T}{1000} \right)^3}
```
```suggestion
C_p(T) =\frac{A0 \cdot \left( \frac{T}{1000} \right)^N}{1 + A1 \cdot \left( \frac{T}{1000} \right) + A2 \cdot \left( \frac{T}{1000} \right)^2 + A3 \cdot \left( \frac{T}{1000} \right)^3},
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+# Specific Heat
+
+For temperature range: 11 ≤ T < 293 K
```
```suggestion
For temperature range: 11 ≤ $T$ < 293 K, the specific heat  is described as:
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+For temperature range: 11 ≤ T < 293 K
+
+\begin{equation}
```
```suggestion
\begin{equation} \label{eq:specific_heat_low}
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+where:
+
+- `C`$_p$`(T)` is **specific heat `[J/g.K]`
+
+- `T` is **temperature `[K]`
+
+| Constant | Value      | Unit                |
+|----------|------------|---------------------|
+| N        | 3.030E+00  | `[1]`               |
+| A0       | 3.103E+02  | `[J/g.K]`           |
+| A1       | -8.815E+00 | `[1]`               |
+| A2       | 1.295E+02  | `[1]`               |
+| A3       | 1.874E+03  | `[1]`               |
```
```suggestion
- $C_p$(T)$ is the specific heat `[J/g.K]`,

- $T$ is the temperature `[K]`,

and the model parameters values provided in [tab:specific_heat_parameters_low].
!table id=tab:specific_heat_parameters_low caption=Parameters values for the specific heat model from [eq:specific_heat_low]. 
| Constant | Value      | Unit                |
|----------|------------|---------------------|
| N        | 3.030E+00  | `[-]`               |
| A0       | 3.103E+02  | `[J/g.K]`           |
| A1       | -8.815E+00 | `[-]`               |
| A2       | 1.295E+02  | `[-]`               |
| A3       | 1.874E+03  | `[-]`               |
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+| A2       | 1.295E+02  | `[1]`               |
+| A3       | 1.874E+03  | `[1]`               |
+
+For temperature range: 293 ≤ T < 3700 K
```
```suggestion
For temperature range: 293 ≤ $T$ < 3700 K, the specific heat is defined as
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+where:
+
+- `C`$_p$`(T)` is **specific heat `[J/g.K]`
+
+- `T` is **temperature `[K]`
+
```
```suggestion
- $C_p$(T)$ is the specific heat `[J/g.K]`,

- $T$ is the temperature `[K]`,

and the model parameters values provided in [tab:specific_heat_parameters_high].
!table id=tab:specific_heat_parameters_high caption=Parameters values for the specific heat model from [eq:specific_heat_high]. 
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+For temperature range: 293 ≤ T < 3700 K
+
+\begin{equation}
```
```suggestion
\begin{equation} \label{eq:specific_heat_high}
```

simopier commented on modules/solid_properties/doc/content/source/materials/TungstenThermalPropertiesMaterial.md:
```diff
@@ -0,0 +1,140 @@
+
+# Density
+
+For temperature range: 5 ≤ T < 3600 K
+\begin{equation}
+\rho(T) = \frac{\rho_{RT}}{\left(1 + \frac{dL}{L_0(T)} \times \frac{1}{100}\right)^3}
+\end{equation}
+
+where:
+
+- `𝜌(T)` is **density `[Kg/m`$^3$`]`
+
+- `𝜌`$_{RT}$`(T)` is **room temperature density `= 19250 [Kg/m`$^3$`]`
+
+- `T` is **temperature `[K]`
+
+and the thermal expansion $dL/L_0(T)$ `[%]` is given by:
+
+\begin{equation}
+\frac{dL(T)}{L_0} = A0 + A1 \cdot \left( \frac{T}{1000} \right) + A2 \cdot \left( \frac{T}{1000} \right)^2 + A3 \cdot \left( \frac{T}{1000} \right)^3
+\end{equation}
+
+For temperature range: 5 ≤ T < 294 K
+
+| Constant | Value      | Unit           |
+|----------|------------|----------------|
+| A0       | -8.529E-02 | `[1]`          |
+| A1       | -9.915E-02 | `[1]`          |
+| A2       | 2.257E+00  | `[1]`          |
+| A3       | -3.157E+00 | `[1]`          |
+
+For temperature range: 294 ≤ T ≤ 3600 K
+
+| Constant | Value      | Unit           |
+|----------|------------|----------------|
+| A0       | -1.400E-01 | `[1]`          |
+| A1       | 4.869E-01  | `[1]`          |
+| A2       | -3.056E-02 | `[1]`          |
+| A3       | 2.234E-02  | `[1]`          |
```
```suggestion
For the temperature range: 5 ≤ $T$ < 3600 K, the density is defined as
\begin{equation} \label{eq:density}
\rho(T) = \frac{\rho_{RT}}{\left(1 + \frac{dL}{L_0(T)} \times \frac{1}{100}\right)^3},
\end{equation}

where:

- $\rho(T)$ is the density `[Kg/m`$^3$`]`,

- $\rho_{RT}(T)$ is the room temperature density, set to 19250 `[Kg/m`$^3$`]`,

- $T$ is the temperature `[K]`,

and the thermal expansion $dL/L_0(T)$ `[%]` is given by:

\begin{equation} \label{eq:thermal_expansion}
\frac{dL(T)}{L_0} = A0 + A1 \cdot \left( \frac{T}{1000} \right) + A2 \cdot \left( \frac{T}{1000} \right)^2 + A3 \cdot \left( \frac{T}{1000} \right)^3.
\end{equation}

The values of the model parameters in [eq:thermal_expansion] are provided in [tab:thermal_expansion_parameters_low] for low temperatures (5 ≤ $T$ < 294 K), and [tab:thermal_expansion_parameters_high] for high temperatures (294 ≤ $T$ ≤ 3600 K).

!table id=tab:thermal_expansion_parameters_low caption=Parameters values for the thermal expansion model from [eq:thermal_expansion] for 5 ≤ $T$ < 294 K. 
| Constant | Value      | Unit           |
|----------|------------|----------------|
| A0       | -8.529E-02 | `[-]`          |
| A1       | -9.915E-02 | `[-]`          |
| A2       | 2.257E+00  | `[-]`          |
| A3       | -3.157E+00 | `[-]`          |

!table id=tab:thermal_expansion_parameters_high caption=Parameters values for the thermal expansion model from [eq:thermal_expansion] for 294 ≤ $T$ ≤ 3600 K. 
| Constant | Value      | Unit           |
|----------|------------|----------------|
| A0       | -1.400E-01 | `[-]`          |
| A1       | 4.869E-01  | `[-]`          |
| A2       | -3.056E-02 | `[-]`          |
| A3       | 2.234E-02  | `[-]`          |
```

simopier reviewed (dismissed):
Looks good to me, but I have quite a few minor changes to the documentation. Will be good to go after that.

meltawila committed:
Address PC's review:
- formatting of doco
- convention for unitless in doco and docstrings

Co-authored-by: Pierre-Clement Simon <pierreclement.simon@gmail.com>

GiudGiud reviewed (changes requested):
@meltawila you ll want to build the documentation locally to make sure it passes

meltawila committed:
Fix building docs errors
